Boolean.CSharp.Test/CoreTests.cs

```diff
@@ -1,24 +1,139 @@
 ﻿using Boolean.CSharp.Main;
 using NUnit.Framework;
+using System.Diagnostics;
+using System.Security.Principal;
+using static System.Runtime.InteropServices.JavaScript.JSType;
 
 namespace Boolean.CSharp.Test
 {
-    [TestFixture]
+
     public class CoreTests
     {
-        private Core _core;
+
 
         public CoreTests()
         {
-            _core = new Core();
 
         }
 
-        [Test]
-        public void TestQuestion1()
+
+        public static IEnumerable<Iaccount> AccountTestCases
+        {
+            get
+            {
+                yield return new Account();
+                yield return new SavingsAccount();
+                yield return new CurrentAccount();
+            }
+        }
+        [TestCaseSource(nameof(AccountTestCases))]
+        public void AddTransaction(Iaccount account)
+        {
+            Account accountCasted ;
+
+            bool res = account.AddTransaction(100f, null);
+
+            List<Transaction> res2 = account.GetTransactions();
+            try
+            {
+
+                accountCasted = (Account)account;
+            }
+            catch (Exception ex) { Assert.Fail("Could not cast to Account class");
+                return;
+            }
+
+            Assert.That(res, Is.True);
+            Assert.That(res2.Count, Is.EqualTo(1));
+            Assert.That(accountCasted.Transactions.Count, Is.EqualTo(1));
+
+        }
+
+        [TestCaseSource(nameof(AccountTestCases))]
```

Comment:
why did you go for this test approach instead of a setup function?

Reply:
I ran into a "compile time constant error" issue when I tried to make a [TestCase(new account(), expectedResult] for each Account type, ie basic Account, SavingsAccount and the CurrentAccount. The TestCaseSource allowed me to itterate over the TestCases in a similar way, without this error. Id still have to make a switch if a used the [TestCase] and setup all the accountTypes in the setup instead. So this seemed more readable to me.

I made sense for me to test each class and child class individually, eventho in this code no implementation differs between them. I was thinking that in a realworld test you want to make sure that all classes behaves as expected, not just the parent.